sw/device/tests/sim_dv/i2c_tx_rx_test.c

```diff
@@ -0,0 +1,209 @@
+                                       kTopEarlgreyPinmuxInselIob11));
+  CHECK_DIF_OK(dif_pinmux_input_select(&pinmux,
+                                       kTopEarlgreyPinmuxPeripheralInI2c0Sda,
+                                       kTopEarlgreyPinmuxInselIob12));
+  CHECK_DIF_OK(dif_pinmux_output_select(&pinmux, kTopEarlgreyPinmuxMioOutIob11,
+                                        kTopEarlgreyPinmuxOutselI2c0Scl));
+  CHECK_DIF_OK(dif_pinmux_output_select(&pinmux, kTopEarlgreyPinmuxMioOutIob12,
+                                        kTopEarlgreyPinmuxOutselI2c0Sda));
+
+  // I2C speed parameters
+  dif_i2c_timing_config_t timing_config = {
+      .lowest_target_device_speed = kDifI2cSpeedFastPlus,
+      .clock_period_nanos = kClockPeriodNanos,
+      .sda_rise_nanos = kI2cRiseFallNanos,
+      .sda_fall_nanos = kI2cRiseFallNanos,
+      .scl_period_nanos = kI2cClockPeriodNanos};
+
+  dif_i2c_config_t config;
+  CHECK_DIF_OK(dif_i2c_compute_timing(timing_config, &config));
+  CHECK_DIF_OK(dif_i2c_configure(&i2c, config));
+  CHECK_DIF_OK(dif_i2c_host_set_enabled(&i2c, kDifToggleEnabled));
+  CHECK_DIF_OK(
+      dif_i2c_set_watermarks(&i2c, kDifI2cLevel30Byte, kDifI2cLevel4Byte));
+  CHECK_DIF_OK(
+      dif_i2c_irq_set_enabled(&i2c, kDifI2cIrqFmtWatermark, kDifToggleEnabled));
+  CHECK_DIF_OK(
+      dif_i2c_irq_set_enabled(&i2c, kDifI2cIrqRxWatermark, kDifToggleEnabled));
+  CHECK_DIF_OK(dif_i2c_irq_set_enabled(&i2c, kDifI2cIrqTransComplete,
+                                       kDifToggleEnabled));
+
+  // randomize variables
+  uint8_t byte_count = rand_testutils_gen32_range(1, 64);
+  uint8_t device_addr = rand_testutils_gen32_range(0, 16);
+  uint8_t expected_data[byte_count];
+  LOG_INFO("Loopback %d bytes with device %d", byte_count, device_addr);
+
+  // controlling the randomization from C side is a bit slow, but might be
+  // easier for portability to a different setup later
+  for (uint32_t i = 0; i < byte_count; ++i) {
+    expected_data[i] = rand_testutils_gen32_range(0, 0xff);
+  };
+
+  // write expected data to i2c device
+  CHECK(!fmt_irq_seen);
+  i2c_testutils_wr(&i2c, device_addr, byte_count, expected_data, false);
+
+  uint8_t tx_fifo_lvl, rx_fifo_lvl;
+
+  // make sure all fifo entries have been drained
+  do {
+    CHECK_DIF_OK(dif_i2c_get_fifo_levels(&i2c, &tx_fifo_lvl, &rx_fifo_lvl));
+  } while (tx_fifo_lvl > 0);
+  CHECK(fmt_irq_seen);
+  fmt_irq_seen = false;
+
+  // read data from i2c device
+  CHECK(!rx_irq_seen);
+  i2c_testutils_rd(&i2c, device_addr, byte_count);
+
+  // make sure all data has been read back
+  do {
+    CHECK_DIF_OK(dif_i2c_get_fifo_levels(&i2c, &tx_fifo_lvl, &rx_fifo_lvl));
+  } while (rx_fifo_lvl < byte_count);
+  CHECK(rx_irq_seen);
+
+  uint8_t byte;
+
+  // make sure every read is the same
+  for (uint32_t i = 0; i < byte_count; ++i) {
+    CHECK_DIF_OK(dif_i2c_read_byte(&i2c, &byte));
+    // LOG_INFO("Expected data 0x%x, read data 0x%x", expected_data[i], byte);
```

Comment:
Delete this line?

Reply:
no i want to keep it for debug purposes when this inevitably goes wrong in the future.

Reply:
what i really wish we had is a `LOG_DEBUG` of sorts that could be turned on based on verbosity of the test. perhaps we have that, i should double check.

Reply:
That would be helpful indeed!

Reply:
i think the main reason to not do the print is because prints are only instantaneous in DV, for all other platforms these things take quite a bit of time... i think i've seen enough hardware corner cases "solved" by print outs because it artificially pushes things apart that I prefer to not have them in fpga / silicon land unless i have to (DV of course is a completely different since they're free). 

that's why i prefer something like `log_debug` that prints out only when needed, without needing the user to go back in and hack-up the code.   I guess in this case i could switch to just the error case like you guys want.

Reply:
Yeah, I'd recommend always printing out error cases, and if you'd like to print out everything only in DV, you can use `kDeviceType` and `kDeviceSimDv` to choose a different path.